src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationPolicyProvider.cs

```diff
@@ -45,7 +45,8 @@ public Task<AuthorizationPolicy> GetDefaultPolicyAsync()
         /// <returns>The named <see cref="AuthorizationPolicy"/>.</returns>
         public virtual Task<AuthorizationPolicy> GetPolicyAsync(string policyName)
         {
-            // MVC relies on DefaultAuthorizationPolicyProvider providing the same policy for the same requests.
+            // MVC relies on this class providing the same policy for the same requests. If it were to provide
+            // different policies per request then the cached policy in MVC could cause undesired access.
```

Comment:
I like @HaoK 's idea. I'd also mention that MVC also relies on it returning synchronously, and also that any changes to either of these behaviors will require us to ship a patch of MVC as well.